front/lib/api/assistant/generation.ts

```diff
@@ -70,46 +70,73 @@ export async function renderConversationForModelMultiActions({
       const actions = removeNulls(m.actions);
 
       // This array is 2D, because we can have multiple calls per agent message (parallel calls).
-      const steps = [] as Array<
-        Array<{ call: FunctionCallType; result: FunctionMessageTypeModel }>
-      >;
+
+      const steps = [] as Array<{
+        content: string | null;
+        actions: Array<{
+          call: FunctionCallType;
+          result: FunctionMessageTypeModel;
+        }>;
+      }>;
 
       if (!excludeActions) {
         for (const action of actions) {
           const stepIndex = action.step;
-          steps[stepIndex] = steps[stepIndex] || [];
-          steps[stepIndex].push({
+          steps[stepIndex] = steps[stepIndex] || {
+            content: null,
+            actions: [],
+          };
+          steps[stepIndex].actions.push({
             call: action.renderForFunctionCall(),
             result: action.renderForMultiActionsModel(),
           });
         }
       }
 
+      for (const content of m.rawContents) {
+        steps[content.step] = steps[content.step] || {
+          content: null,
+          actions: [],
+        };
+        steps[content.step].content = content.content;
```

Comment:
You override here no? There can be multiple content per step in principle no?

Reply:
Same as before, currently it's not technically possible